### ISSUE TITLE
Send the sbatch over directly from string

### ISSUE THREAD

**Issue by bilalshaikh42** [reproducible-biomedical-modeling/CRBM-Viz#1526]
@gmarupilla Does the sbatch need to be sent over as a file? There is a lot of extra work happening here. We have the sbatch file as a string, write it locally, then open up an sftp connection, copy it over, then open an ssh connection, then execute the file. 
Why not just ssh and execute the command directly? 

---

https://github.com/biosimulations/Biosimulations/blob/9c26da60658ead6d6e6fbf2f86ce978623de8ada/biosimulations/apps/dispatch-service/src/app/services/hpc/hpc.service.ts#L50-L55

---

###### This comment was generated by [todo](https://todo.jasonet.co) based on a `todo` comment in 9c26da60658ead6d6e6fbf2f86ce978623de8ada in #1520. cc @biosimulations.

_Originally posted by @todo[bot] in https://github.com/biosimulations/Biosimulations/issues/1520#issuecomment-728630039_

**Comment by gmarupilla** [comment]
I saw the change where you're trying `echo ${sbatchString} | sbatch` to execute the sbatch directly. I'll have to test it because the sbatch command doesn't work if it's not properly indented. The indentation will be an issue if we execute it directly from the string. Still, I'll execute it via string once and verify.

**Comment by bilalshaikh42** [comment]
Yeah I was not sure if it would work. But sbatch man pages do say that it can read the job in from standard input, so there should be a way to have that sent over as one command via ssh.

**Comment by bilalshaikh42** [comment]
@gmarupilla  were you able to test this?

**Comment by gmarupilla** [comment]
I tested it, here are the results:
1. `cat <sbatchfile> | sbatch` --> works
2. sbatchContent=`cat <sbatchFile>`
    echo $sbatchContent | sbatch. --> doesn't work, gives error: `Batch job submission failed: Invalid qos specification`

3. sbatchString=`<Complete SBATCH script as string>`
    --> after hitting enter, gives:` !/bin/bash\n#SBATCH: event not found`
    --> We can make it work by removing ` !/bin/bash` but it won't work as sbatch then, sbatch command needs ` !/bin/bash` at the beginning of sbatch script.

4. sbatchContent=`cat <sbatchFile>`
    echo -e $sbatchContent | sbatch. --> doesn't work, gives error: Batch job submission failed: Invalid qos specification

Seems like echo changes the output while printing it to stdout. @bilalshaikh42 thoughts?

**Comment by bilalshaikh42** [comment]
can we use echo to just write the string to a file, and then provide that
file as input? This should be relatively simple.

**Comment by gmarupilla** [comment]
Tried this:

![image](https://user-images.githubusercontent.com/53095348/99716959-b9676280-2a76-11eb-9011-f84d7f8ed680.png)

This executes the sbatchString in command (after echo) directly, doesn't write to the file.

I tried using quoted SbatchString along with internal quotes escaped using `\`,  which is equivalent to the 3rd case in the above comment. In short, that didn't work either.

What if, we write the batch directly to NFS using FS, then access it directly from HPC instead of writing it through SSH service? NFS is plugged into k8 and HPC both, so filesharing can be done without SSH/SFTP service
